File: examples/1kw-svr.cc
#include <handy/handy.h>

using namespace std;
using namespace handy;

int main(int argc, const char* argv[]) {
    if (argc <= 2) {
        printf("usage: %s <begin port> <end port>\n", argv[0]);
        return 1;
    }
    int begin_port = atoi(argv[1]);
    int end_port = atoi(argv[2]);
    EventBase base;
    vector<TcpServerPtr> svrs(end_port - begin_port);
    for (int i = 0; i < end_port-begin_port; i ++) {
        svrs[i] = TcpServer::startServer(&base, "", begin_port + i);
        svrs[i]->onConnRead([](const TcpConnPtr& con) { con->send(con->getInput()); });
    }
    base.loop();
    info("program exited");
}

File: examples/1kw-svr.py
import sys
import os

if len(sys.argv) < 4:
    print('''usage: %s <begin port> <port count> <process count>''' % sys.argv[0])
    raise SystemExit(1)

begin_port, port_count, process_count = int(sys.argv[1]), int(sys.argv[2]), int(sys.argv[3])
port_per_process = port_count / process_count
for i in range(process_count):
    os.system('./1kw-svr %d %d&' % (begin_port + i * port_per_process, begin_port + (i+1) * port_per_process))
